### PR TITLE
Merge 2.9

### PR DESCRIPTION
**Pull request by wallyworld**
Merge 2.9

#13211 Fix data race (accessing charmInfo) in TestV2CharmExitsApplicationWorker
#13212 Fix retry-provisioning command
#13210 Add note and todo
#13209 Improve readablity of 'no * not found' error message
#13214 Adds remaining authorization objects to ignore
#13215  Use a key-based mutex lock to download resources/agent bins once
#13218 Show application data set by the remote app in show-unit output
#13219 Add impish to the ubuntu series list

Trivial conflicts in imports/code comments.

```
Conflicts:
#       apiserver/facades/client/client/client.go
#       apiserver/facades/schema.json
#       resource/resourceadapters/opener.go
```
## QA steps

See PRs

### CODE REVIEW

File: cmd/clitest/vault.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
+package main
```
Comment (SimonRichardson) [inline cmd/clitest/vault.go:1]: How did this file make it to here?

File: go.mod [hunk 2]
```diff
@@ -45,6 +45,7 @@ require (
 	github.com/hashicorp/golang-lru v0.5.4 // indirect
 	github.com/hashicorp/raft v2.0.0-20200420012049-88ad3b3f0a54+incompatible
 	github.com/hashicorp/raft-boltdb v0.0.0-20171010151810-6e5ba93211ea
+	github.com/im7mortal/kmutex v1.0.1
```
Comment (SimonRichardson) [inline go.mod:48]: Was this approved, didn't see an email to juju-crew.

File: cmd/clitest/vault.go (hunk 1 above)
Comment (wallyworld) [reply cmd/clitest/vault.go:1]: Ah mistake

File: go.mod (hunk 2 above)
Comment (wallyworld) [reply go.mod:48]: It's only a trivial mutex implementation

**Comment by wallyworld** [comment]
!!build!!

File: go.mod (hunk 2 above)
Comment (hpidcock) [reply go.mod:47]: decided to move this to juju/mutex

**Review by hpidcock — APPROVED** [review]

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**